server/src/main/kotlin/com/xebia/functional/xef/server/http/routes/providers/RequestForwardingUtil.kt

```diff
@@ -0,0 +1,138 @@
+package com.xebia.functional.xef.server.http.routes.providers
+
+import com.xebia.functional.xef.server.models.Token
+import io.ktor.client.*
+import io.ktor.client.call.*
+import io.ktor.client.request.*
+import io.ktor.client.statement.*
+import io.ktor.http.*
+import io.ktor.http.content.*
+import io.ktor.server.application.*
+import io.ktor.server.request.*
+import io.ktor.server.response.*
+import io.ktor.util.*
+import  io.ktor.http.URLBuilder
+import io.ktor.util.pipeline.*
+import io.ktor.utils.io.jvm.javaio.*
+
+private const val OAI_URL = "https://api.openai.com"
```

Comment:
In other endpoints, like chatCompletions, we are also accessing this URL. Is this a newly introduced constant? Can we pull this value automatically from the OpenAi lib deps?

Reply:
The constant used by chatCompletions additionally holds the path /v1/ and is a hardcode in the endpoint's file.
This url should only contain the base url, as the path and parameters are copied automatically from the incoming request with the new utility functions I introduced in this PR.

It is not desirable to use this variable from the Aallam library, as the server shouldn't have the library as a dependency right. We could create a variable in xef-core like DEFAULT_OPENAI_URL that can be used by the xef-server. Furthermore, I just looked it up, the Aallam's url to OpenAI contains the v1 path too.